requirements-optional.txt

```diff
@@ -1,5 +1,5 @@
 yapf<0.33
 toml<0.11
-codecov
 pytest-cov
 ipykernel
```

Comment:
de we want the ipykernel as a dependency? It's only really used for the jupyter notebooks which some has to setup themselves, if they want to use them, IMO.

Reply:
I would keep it, as sometimes it is useful

Reply:
We can just install it as part of the CI where we need it. I feel like it's a bit bm to install software on a users plattform that is not strictly required.

Reply:
I think at some point we had some issue with @superlopuh where he could not run a notebook without this?

Reply:
Yes, this is a notebook dependency, so you can't run them without this. However, I am arguing that this is not a dependency of our xdsl developer setup, but rather a dependency of running any jupyter notebook, so something that should be installed while installing jupyter notebooks themselves and not just ours.

But I am fine with leaving it as is. It was just something I wasn't sure about :)

Reply:
Ideally we would like users to be able to run notebooks just by installing the base reqs..